src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -829,6 +829,34 @@ public static void DeserializeUserDefinedDictionaryThrows()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Parse<IImmutableDictionaryWrapper>(json));
         }
 
+        [Fact]
+        public static void Regression38565_Serialize()
+        {
+            var value = new Regression38565_Parent()
+            {
+                Test = "value1",
+                Child = new Regression38565_Child()
+            };
+
+            var actual = JsonSerializer.ToString(value, new JsonSerializerOptions { IgnoreNullValues = true });
+            Assert.Equal("{\"Test\":\"value1\",\"Child\":{}}", actual);
+        }
+
+        [Fact]
+        public static void Regression38565_Deserialize()
+        {
+            // The json contains nulls to ensure that the dictionary is properly closed
+
+            var json = "{\"Test\":\"value1\",\"Dict\":null,\"Child\":{\"Dict\":null}}";
+            var actual = JsonSerializer.Parse<Regression38565_Parent>(json);
+
+            Assert.True(actual.Test == "value1");
```

Comment:
nit: Use Assert.Equals() here

```suggestion
            Assert.True("value1", actual.Test);
```

Reply:
Fixed. I had just copied the test as is from the bug.